### ISSUE TITLE
noImportCycles - false positive on CLI, but fine in VS Code and Playground

### ISSUE THREAD

**Issue by MisterJimson** [biomejs/biome#6569]
### Environment information

```bash
CLI:
  Version:                      2.0.5
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-ghostty
  JS_RUNTIME_VERSION:           v22.15.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/9.15.9

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  true
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noEmptyTypeParameters
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/useArrowFunction
    complexity/useFlatMap
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    correctness/noChildrenProp
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useIsNan
    correctness/useValidForDirection
    correctness/useValidTypeof
    nursery/noImportCycles
    performance/noDynamicNamespaceImportAccess
    security/noBlankTarget
    security/noDangerouslySetInnerHtml
    security/noDangerouslySetInnerHtmlWithChildren
    security/noGlobalEval
    style/noDescendingSpecificity
    style/useArrayLiterals
    style/useConst
    style/useDeprecatedReason
    style/useExponentiationOperator
    style/useExportType
    style/useImportType
    style/useLiteralEnumMembers
    style/useShorthandFunctionType
    suspicious/noApproximativeNumericConstant
    suspicious/noAsyncPromiseExecutor
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConstEnum
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noEmptyBlock
    suspicious/noEmptyInterface
    suspicious/noExplicitAny
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShorthandPropertyOverrides
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useDefaultSwitchClauseLast
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useIsArray
    suspicious/useNamespaceKeyword

Workspace:
  Open Documents:               0
```

### Rule name

noImportCycles

### Playground link

https://biomejs.dev/playground/?lintRules=noImportCycles&code=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIABlAHgAYQBtAHAAbABlADEAKAApACAAewAKACAAIAByAGUAdAB1AHIAbgAgADEAOwAKAH0ACgAKAGUAeABwAG8AcgB0ACAAZgB1AG4AYwB0AGkAbwBuACAAZQB4AGEAbQBwAGwAZQAyACgAKQAgAHsACgAgACAAcgBlAHQAdQByAG4AIAAyADsACgB9AAoACgBlAHgAcABvAHIAdAAgACoAIABhAHMAIABFAHgAYQBtAHAAbABlACAAZgByAG8AbQAgACcALgAvAHQAZQBzAHQAJwA7AA%3D%3D

### Expected result

The follow type of code triggers noImportCycles on biome cli, but not in the editor or playground.
```ts
export function example1() {
  return 1;
}

export function example2() {
  return 2;
}

export * as Example from './test';
```

Example:
```
packages/core/src/event-bus/client.ts:181:27 lint/nursery/noImportCycles ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ This import is part of a cycle.

    179 │ }
    180 │
  > 181 │ export * as EventBus from './client';
        │                           ^^^^^^^^^^
    182 │

  ℹ This import resolves to packages/core/src/event-bus/client.ts
        ... which imports packages/core/src/event-bus/client.ts
        ... which is the file we're importing from.
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
Usually, if it doesn't show up in the playground, it's likely that it's already been fixed. The playground is based on the latest commit on `main`.

However, I'm not too sure of the status of rules that require multi file analysis in the playground.

**Comment by ematipico** [comment]
Indeed, project rules don't work in the playground. Please @MisterJimson provide us a repository to test

**Comment by arendjr** [comment]
I'm actually on the fence whether this should be considered a false positive at all. Files that either accidentally or intentionally import/export from themselves _are_ creating a cycle, albeit a very small one. In many cases I think people want to be alerted about this, so I think this isn't an issue, although you could open a discussion whether we should have an option for this case.

What actually worries me more is why it's not being reported in your IDE 🤔

**Comment by MisterJimson** [comment]
FYI This case isn't marked as a cycle in eslint. And oxc has the same bug and are fixing it: https://github.com/oxc-project/oxc/issues/11646

We are switching to whichever project fixes it first.

**Comment by ematipico** [comment]
FYI @arendjr, if we want to keep the existing behaviour, we should change the rule metadata and mark it as inspired. Currently, it is marked as "the same as".

**Comment by dyc3** [comment]
I just tried it in both node and bun. Both allow the file to import itself without blowing up.

Although I tend to agree that this would be a code smell. I imagine the goal of the original rule was to alert for things that would crash and burn.

**Comment by MisterJimson** [comment]
Does anyone have a suggested alternative approach that preserves similar syntax?

```ts
export function example1() {
  return 1;
}

export function example2() {
  return 2;
}

export * as Example from './test';

// usage example
import { Example } from '@package/test'

const value = Example.example2();
```

The goal is to have namespaced functions.

**Comment by dyc3** [comment]
You could put all the functions in a static class, and then just export the class.

**Comment by MisterJimson** [comment]
That's an idea. Would need to disable https://biomejs.dev/linter/rules/no-static-only-class/  though.

The other thing to consider is tree shaking. If it's purely an import style like I have now, I believe fully unused functions are still able to be removed. Would the same be true for a static class? Not sure...

**Comment by dyc3** [comment]
In that case, I think it makes sense to not flag these cases where the file imports itself.

**Comment by ematipico** [comment]
I think so too, we shouldn't flag this case

**Comment by emilyinure** [comment]
I can take this one as well
